packages/core/src/services/TransactionsService.ts

```diff
@@ -18,4 +22,25 @@ export class TransactionsService implements ITransactionsService {
 
     return provider.waitForTransaction(txHash);
   }
+
+  public async waitForEvent(
+    filter: ethers.EventFilter,
+    durationMs: number
+  ): Promise<ethers.providers.Log | null> {
+    const provider = new ethers.providers.Web3Provider(
+      this._providers.ethereumProvider
+    );
+
+    return new Promise((resolve) => {
+      const timeout = setTimeout(() => {
```

Comment:
Can I check my understanding here.

We wait for event, __for some amount of time__. Then fail if it hasn't occurred?

What happens if an event has already happened before `provider.once`, does it return immediately?

Reply:
> We wait for event, for some amount of time. Then fail if it hasn't occurred?

correct

> does it return immediately?

also correct. if an event matching the provided filter has happened before that event listener is registered, it will get triggered